### ISSUE TITLE
Disabling autodoc_docstring_signature causes unrelated tests to fail

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#718]
When trying to backport the autodoc_docstring_signature feature, I noticed that there are unexpected issues when running the sphinx unit tests when autodoc_docstring_signature is False.

Change

```
    app.add_config_value('autodoc_docstring_signature', True, True)
```

in sphinx/ext/autodoc.py to

```
    app.add_config_value('autodoc_docstring_signature', False, True)
```

to get

```
======================================================================
FAIL: test_autodoc.test_generate
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/pymodules/python2.6/nose/case.py", line 187, in runTest
    self.test(*self.arg)
  File "/home/nikratio/tmp/sphinx/tests/test_autodoc.py", line 486, in
test_generate
    assert_processes([('function', 'time.asctime')], 'function', 'asctime')
  File "/home/nikratio/tmp/sphinx/tests/test_autodoc.py", line 351, in
assert_processes
    assert_works(objtype, name, **kw)
  File "/home/nikratio/tmp/sphinx/tests/test_autodoc.py", line 345, in
assert_works
    assert len(_warnings) == 0, _warnings
AssertionError: ["error while formatting arguments for time.asctime:
<method-wrapper '__init__' of builtin_function_or_method object at
0x7f1f5b049a28> is not a Python function"]

----------------------------------------------------------------------
```

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/718
- Originally reported by: Nikolaus Rath
- Originally created at: 2011-06-15T19:07:04.195

**Comment by shimizukawa** [comment]
_From Nikolaus Rath on 2011-06-29 00:20:18+00:00_

Jakub Wilk suggested that line 915 of sphinx/ext/autodoc.py looks very fishy:

{{{
#!python
        if inspect.isbuiltin(self.object) or \
               inspect.ismethoddescriptor(self.object):
            # cannot introspect arguments of a C function or method
            pass
}}}

and indeed, replacing `pass` by `return` fixes the problem. Note, however, that I have no idea what's going on here exactly. I just blindly changed this and noticed that all the unit tests work fine after that.

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2011-09-21 07:18:51+00:00_

Fix #718: "pass" is not a good way to exit a function.

→ <<cset 44f396488c318b1b686d928e516b35be12b052d6>>
